cypress/support/helpers/visitPage.js

```diff
@@ -1,11 +1,14 @@
+import { dataEndpointOverride } from './onDemandRadioTv';
+
 export default (path, pageType) => {
   const expectedContentType = 'text/html';
   const isErrorPage = pageType.includes('error');
   const expectedStatus = isErrorPage ? 404 : 200;
   const failOnStatusCode = !isErrorPage;
+  const newPath = `${path}${dataEndpointOverride()}`;
+  cy.testResponseCodeAndType(newPath, expectedStatus, expectedContentType);
```

Comment:
Is the purpose of this to ensure that the page data is also accessible?

Reply:
It is mainly so that this function can also be used on test for pages that need live data, like the pages with schedules, and need the override to be added to it. I used it in the testResponseCodeAndType so it is checking with that same URL being visited, but maybe it is better to check the response without the override? What do you think?

Reply:
Eek not sure. While I like the idea, this could end up adding time to every single test, because as well as visiting the page, we are also checking the json endpoint. In some tests, we check the json endpoint again. I think we probably need to remove the logic from this common location, and perhaps add a new helper / check in our on demand radio / tv tests to check the endpoint. It could even be a completely separate test.